### PR TITLE
Rename DimensionUnit in FormRecognizer

### LINKED ISSUE #11775: Rename DimensionUnit to LengthUnit
- [ ] Remove ExpandableStringEnums `values` method as it is not needed.

- [ ] Rename `DimensionUnit` to `LengthUnit` [Cross language consistency]

### PR DESCRIPTION
**Pull request by samvaity**
Closes #11775

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Release History
 
 ## 1.0.0-beta.3 (Unreleased)
+- Rename `DimensionUnit` to `LengthUnit`
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:4]: consider add the class where the property lives

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Release History
 
 ## 1.0.0-beta.3 (Unreleased)
+- Rename enum type `DimensionUnit` to `LengthUnit` on `FormPage`
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:4]: nit: I don't think you need to include "on `FormPage`"

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:4]: that was me asking. From the top of my head I couldn't remember where the property lived.
By having the type, our users can easily know if it affects them or not.

**Comment by samvaity** [comment]
/ azp run java - formrecognizer - ci

**Pull request merged**